### PR TITLE
attemptSilentLogin feature

### PR DESCRIPTION
**Pull request by adamjmcgrath**
### Description

Add `attemptSilentLogin` MW and `attemptSilentLogin` config option to attempt login with `prompt=none` the first time a user visits a route.

### References

https://auth0.com/docs/api-auth/tutorials/silent-authentication

### Testing

- [x] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [x] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not `master`

### CODE REVIEW

File: lib/context.js [hunk 1]
```diff
@@ -227,6 +237,8 @@ class ResponseContext {
       returnURL = urlJoin(config.baseURL, returnURL);
     }
 
+    cancelSilentLoginAttempts(req, res);
```
Comment (adamjmcgrath) [inline lib/context.js:240]: see: https://github.com/auth0/express-openid-connect/blob/abef7c5ccb8911d57a32d8428bb9b63599359222/test/attemptSilentLogin.tests.js#L129-L146

File: middleware/requiresAuth.js [hunk 2]
```diff
@@ -17,7 +17,7 @@ async function requiresLoginMiddleware(requiresLoginCheck, req, res, next) {
   }
 
   if (requiresLoginCheck(req)) {
-    if (!res.oidc.errorOnRequiredAuth) {
+    if (!res.oidc.errorOnRequiredAuth && req.accepts('html')) {
```
Comment (adamjmcgrath) [inline middleware/requiresAuth.js:20]: Also added the `html` check to `requiresAuth`

File: middleware/attemptSilentLogin.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+const debug = require('../lib/debug');
+const COOKIES = require('../lib/cookies');
+
+const COOKIE_NAME = 'silentLoginAttempted';
+
+const cancelSilentLoginAttempts = (req, res) =>
+  res.cookie(COOKIE_NAME, true, {
+    httpOnly: true,
+    secure: req.secure,
```
Comment (panva) [inline middleware/attemptSilentLogin.js:9]: this ought to take into consideration an explicit configuration, similar to what we do in appSession.js

```
secure: typeof cookieConfig.secure === 'boolean'
          ? cookieConfig.secure
          : req.secure
```

File: middleware/auth.js [hunk 4]
```diff
@@ -148,6 +149,10 @@ module.exports = function (params) {
         'see and apply `requiresAuth` middlewares to your protected resources'
     );
   }
+  if (config.attemptSilentLogin) {
+    debug.trace("silent login will be attempted on user's initial request");
```
Comment (panva) [inline middleware/auth.js:153]:
```suggestion
    debug.trace("silent login will be attempted on end-user's initial HTML request");
```

File: lib/context.js [hunk 5]
```diff
@@ -127,6 +130,13 @@ class ResponseContext {
...
+  silentLogin(options) {
+    return this.login({
+      ...options,
+      prompt: 'none',
+    });
+  }
+
   async login(options = {}) {
     let { config, req, res, next, transient } = weakRef(this);
     next = cb(next).once();
```
Comment (adamjmcgrath) [inline lib/context.js:142]: TODO: Clear cookie on login

File: middleware/attemptSilentLogin.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
+const debug = require('../lib/debug');
+const COOKIES = require('../lib/cookies');
+
+const COOKIE_NAME = 'silentLoginAttempted';
```
Comment (adamjmcgrath) [inline middleware/attemptSilentLogin.js:4]: `skipSilentLogin`

File: middleware/attemptSilentLogin.js (hunk 3 above)
Comment (adamjmcgrath) [reply middleware/attemptSilentLogin.js]: 6c1cb89

File: middleware/auth.js (hunk 4 above)
Comment (adamjmcgrath) [reply middleware/auth.js]: 6c1cb89

File: lib/context.js (hunk 5 above)
Comment (adamjmcgrath) [reply lib/context.js:140]: 6c1cb89

File: middleware/attemptSilentLogin.js (hunk 6 above)
Comment (adamjmcgrath) [reply middleware/attemptSilentLogin.js]: 6c1cb89

**Pull request merged**